pkgdb/src/buildenv/assets/etc/profile.d/0800_cuda.sh

```diff
@@ -0,0 +1,41 @@
+# shellcheck shell=bash
+export _coreutils="@coreutils@"
+export _gnused="@gnused@"
+# ============================================================================ #
+#
+# Setup CUDA
+#
+# ---------------------------------------------------------------------------- #
+
+# Only run if FLUX_ENV_ENABLE_CUDA feature flag is set
+activate_cuda(){
+  if [[ "$FLOX_ENV_ENABLE_CUDA" != 1 ]]; then
+    return 1
+  fi
+
+  devices=$( "$_coreutils/bin/ls" -1 /dev/nvidia* 2>/dev/null | "$_coreutils/bin/wc" -l )
+  if [[ "$devices" == 0 ]]; then
+    return 2
+  fi
+
+  if ! ldconfig -p >/dev/null 2>&1 ; then
+    return 3
+  fi
+
+  NEW_FLOX_ENV_LIB_DIRS="$(ldconfig -p | \
```

Comment:
As I understand it, this adds the entire contents of the ld cache to the NEW_FLOX_ENV_LIB_DIRS variable - could this be too greedy?

Reply:
Yes, quite greedy. I'm not thrilled with it. I'm open to alternatives.

Reply:
Understood. It does seem to be effective on my system 👍